examples/stm32h7/src/bin/mco.rs

```diff
@@ -0,0 +1,32 @@
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+
+#[path = "../example_common.rs"]
+mod example_common;
+use embassy::executor::Spawner;
+use embassy::time::{Duration, Timer};
+use embassy_stm32::gpio::{Level, Output, Speed};
+use embassy_stm32::rcc::{Mco, Mco1Source, McoClock};
+use embassy_stm32::Peripherals;
+use embedded_hal::digital::v2::OutputPin;
+use example_common::*;
+
+#[embassy::main]
+async fn main(_spawner: Spawner, p: Peripherals) {
+    info!("Hello World!");
+
+    let mut led = Output::new(p.PB14, Level::High, Speed::Low);
+
+    let _mco = Mco::new(p.PA8, Mco1Source::Hsi, McoClock::Divided(2));
```

Comment:
There's no owned singleton for RCC MCO1/MCO2, so you could create a second `Mco` instance with the same Mco 1/2 that would break the previously created one.

Could you add RCC_MCO1, RCC_MCO2 peripheral singletons (they're created [here](https://github.com/embassy-rs/embassy/blob/fc75e0ec573267d7b97698ea140e1533e356e012/embassy-stm32/build.rs#L19)) and make `Mco` require them?

Reply:
May be also able to create it via YAML like the VREFBUF?

https://github.com/embassy-rs/stm32-data/blob/main/data/extra/family/STM32L4.yaml

Reply:
But maybe not, since it doesn't have an address per se?

Reply:
@Dirbaio I designed it around the singletons for the pins (or at least I hope that I haven't messed it up), I agree that this is suboptimal.

Could you please give me more pointers about defining a custom singleton? We can discuss it on matrix so that we do not clutter the discussion here.

Reply:
@bobmcwhirter it's not an actual peripheral so it should go in embassy instead of stm32-data. -data should have data for the "unopinionated ground truth". Adding extra singletons because it helps making a safe api is a nopinionated decision made by embassy-stm32  :) 

@matoushybl you can check the RCC version with `peripheral_versions!` and append the right singletons to the [`singletons`](https://github.com/embassy-rs/embassy/blob/fc75e0ec573267d7b97698ea140e1533e356e012/embassy-stm32/build.rs#L29) vec, like we already do for gpio, exti, dma.